### ISSUE TITLE
SPIKE: Use SNYK cli for securiy scan of upstream repos

### ISSUE THREAD

**Issue by vaibhavjainwiz** [opendatahub-io/kserve#53]
following are the steps to run a Snyk code scan using CLI.
Snyk CLI installation steps can be found at [1]

Please let me know in case you need any additional information.

Step 1 - Authenticate to Snyk via CLI, will redirect you to browser, click "Authenticate" button, come back to CLI

$ snyk auth

Now redirecting you to our auth page, go ahead and log in,
and once the auth is complete, return to this prompt and you'll
be ready to start using snyk.

If you can't wait use this url:
https://app.snyk.io/login?token=e6a20279-5532-4d8f-a055-946a6da8b2da&utm_medium=cli&utm_source=cli&utm_campaign=CLI_V1_PLUGIN&utm_campaign_content=1.1161.0&os=linux&docker=false

Your account has been authenticated. Snyk is now ready to be used.

$

Step 2 - Clone the repo that you wish to scan and move into it

$ git clone https://github.com/ansible/ansible.git
Cloning into 'ansible'...
remote: Enumerating objects: 601185, done.
remote: Counting objects: 100% (848/848), done.
remote: Compressing objects: 100% (471/471), done.
remote: Total 601185 (delta 458), reused 676 (delta 336), pack-reused 600337
Receiving objects: 100% (601185/601185), 235.99 MiB | 11.21 MiB/s, done.
Resolving deltas: 100% (400419/400419), done.
$
$ ls
ansible
$
$ cd ansible/
$
$ pwd
/tmp/x/ansible
$ 

Step 3 - Scan the repo using command below, you can get the org id from Settings in Snyk UI

All results are visible on the CLI directly, can apply more filters, check appropriate CLI options.

$ snyk code test --org="25b9c761-8536-47b4-95de-0d06931deeca"

Testing /tmp/x/ansible ...

 ✗ [Low] Arbitrary File Write via Archive Extraction (Tar Slip)
   Path: test/lib/ansible_test/_internal/commands/sanity/validate_modules.py, line 162
   Info: Calling extractall to extract all files from a tar file without sanitization. This may result files outside destination directory to be overwritten, resulting in an arbitrary file write.

<< snip >>

 ✗ [Low] Deserialization of Untrusted Data
   Path: test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py, line 97
   Info: Unsanitized input from a command line argument flows into yaml.load, where it is used to deserialize an object. This may result in an Unsafe Deserialization vulnerability.

 ✗ [High] Regular Expression Denial of Service (ReDoS)
   Path: hacking/azp/incidental.py, line 192
   Info: Unsanitized user input from a command line argument flows into re.search, where it is used to build a regular expression. This may result in a Regular expression Denial of Service attack (reDOS).

<< snip >>

✔ Test completed

Organization:      25b9c761-8536-47b4-95de-0d06931deeca
Test type:         Static code analysis
Project path:      /tmp/x/ansible

Summary:

  143 Code issues found
  4 [High]   41 [Medium]   98 [Low]

**Comment by vaibhavjainwiz** [comment]
Snyk also have a tool (in beta) which can be used for comparison purpose. Feel free to try it out
https://docs.snyk.io/snyk-api/other-tools/tool-snyk-delta

**Comment by heyselbi** [comment]
Duplicate of an issue: setting up SNYK on upstream repos
